lib/puppet/application/device.rb

```diff
@@ -219,20 +238,36 @@ def main
           Puppet[:vardir] = ::File.join(Puppet[:devicedir], device.name)
           Puppet[:certname] = device.name
 
-          # this will reload and recompute default settings and create the devices sub vardir, or we hope so :-)
-          Puppet.settings.use :main, :agent, :ssl
-
```

Comment:
I think we need to do this in the `apply` case also? What happens if you delete `~/.puppetlabs` and try to run `puppet device --apply` with this PR?